packages/neon/neon/lib/src/settings/models/storage.dart

```diff
@@ -2,32 +2,87 @@ import 'package:meta/meta.dart';
 import 'package:nextcloud/ids.dart';
 import 'package:shared_preferences/shared_preferences.dart';
 
+/// Storage interface used by `Option`s.
+///
+/// Mimics the interface of [SharedPreferences].
+///
+/// See:
+///   * [SingleValueStorage] for a storage that saves a single value.
+///   * [AppStorage] for a storage that fully implements the [SharedPreferences] interface.
+///   * [NeonStorage] that manages the storage backend.
 @internal
 abstract interface class SettingsStorage {
+  /// {@template NeonStorage.getString}
+  /// Reads a value from persistent storage, throwing an exception if it's not a String.
+  /// {@endtemplate}
   String? getString(final String key);
 
+  /// {@template NeonStorage.setString}
+  /// Saves a string [value] to persistent storage in the background.
+  ///
+  /// Note: Due to limitations in Android's SharedPreferences,
+  /// values cannot start with any one of the following:
+  ///
+  /// - 'VGhpcyBpcyB0aGUgcHJlZml4IGZvciBhIGxpc3Qu'
+  /// - 'VGhpcyBpcyB0aGUgcHJlZml4IGZvciBCaWdJbnRlZ2Vy'
+  /// - 'VGhpcyBpcyB0aGUgcHJlZml4IGZvciBEb3VibGUu'
```

Comment:
Can we just point to some other docs instead of noting this down ourselves?

Reply:
I just copied the doc comment from the shared prefs package (smae for all other doc comments here).
I don't it is a problem having them here.